### ISSUE TITLE
Python.exe file closes when there should be playing mp3 file

### ISSUE THREAD

**Issue by fritzl123321** [clinton-hall/nzbToMedia#1612]
So i wanted to code a little program that plays an mp3 file when something is happening. Nothing really crazy. So i converted it to an exe file with auto-py-to-exe bla bla bla.. and boom, everytime it should play the mp3 file, the console closes and the program is finished. any ideas of fixing this? thanks!

**Comment by labrys** [comment]
This isn't exactly the place for that, it would be more of a question for stack overflow.  However, if you post some code so I can see what you're actually trying I'll take a look at it.

**Comment by fritzl123321** [comment]
> This isn't exactly the place for that, it would be more of a question for stack overflow. However, if you post some code so I can see what you're actually trying I'll take a look at it.

Sorry, I am a newbie haha, will do that in future!

So:

import PIL.ImageGrab
import time
from playsound import playsound

def getHealth():
    PIL.ImageGrab.grab().size

    global health
    health = PIL.ImageGrab.grab().load()[775, 1050]


def getMana():
    PIL.ImageGrab.grab().size

    global Q
    global W
    global E
    global R

    Q = PIL.ImageGrab.grab().load()[808, 993]
    W = PIL.ImageGrab.grab().load()[852, 993]
    E = PIL.ImageGrab.grab().load()[896, 993]
    R = PIL.ImageGrab.grab().load()[940, 993]


def checkHealth():
    getHealth()

    if health[0] == 192 and health[1] == 252 and health[2] == 250:
        playsound('health.mp3')


def checkMana():
    getMana()

    if (Q[0] >= 0 and Q[0] <= 5) and (Q[1] >= 25 and Q[1] <= 32) and (Q[2] >= 27 and Q[2] <= 35):
        print("Q\n")
        playsound('mana .mp3')

    if (W[0] >= 0 and W[0] <= 5) and (W[1] >= 25 and W[1] <= 32) and (W[2] >= 27 and W[2] <= 35):
        print("W\n")
        playsound('mana.mp3')

    if (E[0] >= 0 and E[0] <= 5) and (E[1] >= 25 and E[1] <= 32) and (E[2] >= 27 and E[2] <= 35):
        print("E\n")
        playsound('mana.mp3')

    if (R[0] >= 0 and R[0] <= 5) and (R[1] >= 25 and R[1] <= 32) and (R[2] >= 27 and R[2] <= 35):
        print("R\n")
        playsound('mana.mp3')


def main():
    while True:
        checkHealth()
        checkMana()
        time.sleep(1)


main()


basically nothing's important, except the playsound function which is the reason of the program basically

**Comment by labrys** [comment]
What errors are you getting?  Please post the actual error message.  My initial guess is that the mp3 is in a different folder.  My second guess is the playsound library, as I have heard it has had issues in the past.  The traceback could give us a better indication.

**Comment by labrys** [comment]
I'm gonna close this thread since its off topic, but discussion can continue here for now.  If it gets too long, we can take it to another forum.

**Comment by fritzl123321** [comment]
> What errors are you getting? Please post the actual error message. My initial guess is that the mp3 is in a different folder. My second guess is the playsound library, as I have heard it has had issues in the past. The traceback could give us a better indication.

So when it should play music, its just closing leaving no message at all. I put the music into the right place in my opinion, i can at least run the program in pycharm. I will try to find another module that will work to play mp3 and then I'm going to tell you what happened.

**Comment by labrys** [comment]
Can you run it from the command prompt and does the music play as expected?

**Comment by labrys** [comment]
I've created a repo at https://github.com/labrys/fritzlsound we can discuss it further there and I can show you some code improvements.

**Comment by fritzl123321** [comment]
> Can you run it from the command prompt and does the music play as expected?

dumb question but what do you mean with command prompt?

**Comment by fritzl123321** [comment]
> I've created a repo at https://github.com/labrys/fritzlsound we can discuss it further there and I can show you some code improvements.

yeah that would be awesome!
